### DISCUSSION TITLE
How can we fetch parallel value from command line to JS file

### DISCUSSION THREAD

**Discussion by windyliu1118** [getgauge/gauge#1951] [General]
Currently, in my code, I need to run different logic based on parallel running or non-parallel running.
So when I run the command `gauge run  --env staging --tags "p1" --parallel -n=2 specs`, how can I fetch the parallel running parameter from the command line, and then use it in Js file to determine which code to be executed?

**Comment by zabil** [comment]
The easiest way would be passing an extra environment variable when running tests in parallel. For example.

```
PARALLEL=true gauge run --env staging --tags "p1" --parallel -n=2 specs
```

You can then read this environment variable in your step implementation.

```
if(process.env.PARALLEL) {
 ...
}
```

**Reply by windyliu1118** [reply]
Got it, thanks @zabil
